libvast/vast/concept/parseable/numeric/integral.hpp

```diff
@@ -32,32 +33,44 @@ bool parse_sign(Iterator& i) {
 
 } // namespace detail
 
-template <
-  class T,
-  int MaxDigits = std::numeric_limits<T>::digits10 + 1,
-  int MinDigits = 1,
-  int Radix = 10
->
+template <class T,
+          // Note that it is fine to have this specific for base 10, as other
+          // bases must be specified as a later template parameters anyways.
+          int MaxDigits = std::numeric_limits<T>::digits10 + 1,
+          int MinDigits = 1, int Radix = 10>
 struct integral_parser
   : parser<integral_parser<T, MaxDigits, MinDigits, Radix>> {
-  static_assert(Radix == 10, "unsupported radix");
+  static_assert(Radix == 10 || Radix == 16, "unsupported radix");
   static_assert(MinDigits > 0, "need at least one minimum digit");
   static_assert(MaxDigits > 0, "need at least one maximum digit");
   static_assert(MinDigits <= MaxDigits, "maximum cannot exceed minimum");
 
   using attribute = T;
 
   static bool isdigit(char c) {
-    return c >= '0' && c <= '9';
+    if constexpr (Radix == 10)
+      return c >= '0' && c <= '9';
+    else if constexpr (Radix == 16)
+      return std::isxdigit(c);
+    else
+      static_assert(detail::always_false_v<decltype(Radix)>, "unsupported "
```

Comment:
Why do you need `decltype` here?

Reply:
Radix is an integer value, not a type, and the latter is expected by the type trait `detail::always_false`.